File: code/src/main/java/com/six/finger/leetcode/two/TwoSum.java
package com.six.finger.leetcode.two;

import java.util.HashMap;
import java.util.Map;

/**
 * @author wangbin3@joyy.sg
 * @version 1.0
 * @date 2021/11/4 20:53
 */
public class TwoSum {


    public int[] twoSum(int[] nums, int target) {

        int[] res = new int[2];

        Map<Integer,Integer> map=new HashMap<>();

        for (int i = 0; i <nums.length; i++) {
            if (map.containsKey(target-nums[i])){
                res[0]=i;
                res[i]=map.get(target-nums[i]);
                return res;
            }else {
                map.put(target-nums[i],i);
            }
        }
        return res;


    }
}

File: code/src/main/java/com/six/finger/leetcode/go/twoSum.go
package main

func main() {

}

func twosum(nums []int, target int) (res []int) {

	for i, num := range nums {
		for j := i + 1; j < len(nums); j++ {
			if num+nums[j] == target {
				res := make([]int, 2, 2)
				res[0] = i
				res[1] = j
				return res
			}

		}
	}
	return nil

}

func twosum1(nums []int, target int) (res []int) {

	maps := make(map[int]int)
	for i, num := range nums {
		value, ok := maps[target-num]
		if ok {
			return []int{value, i}
		} else {
			maps[num] = i
		}

	}

	return nil

}
